oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/Utils.java

```diff
@@ -1,130 +1,92 @@
-                                                      @NotNull final String containerName) throws DataStoreException {
-        return getBlobContainer(connectionString, containerName, null);
-    }
+            BlobServiceClient blobServiceClient = new BlobServiceClientBuilder()
+                    .connectionString(connectionString)
+                    .retryOptions(retryOptions)
+                    .buildClient();
+            blobContainerClient = blobServiceClient.getBlobContainerClient(containerName);
+            return blobContainerClient;
 
-    public static CloudBlobContainer getBlobContainer(@NotNull final String connectionString,
-                                                      @NotNull final String containerName,
-                                                      @Nullable final BlobRequestOptions requestOptions) throws DataStoreException {
-        try {
-            CloudBlobClient client = (
-                    (null == requestOptions)
-                            ? Utils.getBlobClient(connectionString)
-                            : Utils.getBlobClient(connectionString, requestOptions)
-            );
-            return client.getContainerReference(containerName);
-        } catch (InvalidKeyException | URISyntaxException | StorageException e) {
+        } catch (Exception e) {
             throw new DataStoreException(e);
         }
     }
 
-    public static void setProxyIfNeeded(final Properties properties) {
+    public static boolean isProxyNeeded(final Properties properties) {
         String proxyHost = properties.getProperty(AzureConstants.PROXY_HOST);
         String proxyPort = properties.getProperty(AzureConstants.PROXY_PORT);
 
-        if (!Strings.isNullOrEmpty(proxyHost) &&
-            Strings.isNullOrEmpty(proxyPort)) {
-            int port = Integer.parseInt(proxyPort);
-            SocketAddress proxyAddr = new InetSocketAddress(proxyHost, port);
-            Proxy proxy = new Proxy(Proxy.Type.HTTP, proxyAddr);
-            OperationContext.setDefaultProxy(proxy);
-        }
+        return !Strings.isNullOrEmpty(proxyHost) && Strings.isNullOrEmpty(proxyPort);
     }
 
-    public static RetryPolicy getRetryPolicy(final String maxRequestRetry) {
-        int retries = PropertiesUtil.toInteger(maxRequestRetry, -1);
-        if (retries < 0) {
+    public static RequestRetryOptions getRetryOptions(final String maxRequestRetryCount) {
+        int retries = PropertiesUtil.toInteger(maxRequestRetryCount, -1);
+        if(retries < 0) {
             return null;
         }
+
         if (retries == 0) {
-            return new RetryNoRetry();
+            return new RequestRetryOptions(RetryPolicyType.FIXED, 1, 1, null, null, null);
         }
-        return new RetryExponentialRetry(RetryPolicy.DEFAULT_CLIENT_BACKOFF, retries);
+        return new RequestRetryOptions(RetryPolicyType.EXPONENTIAL, retries, (int) Duration.ofSeconds(3).toMillis(), null, null, null);
```

Comment:
`tryTimeoutInSeconds` should not be hardcoded. it should use the `AzureConstants.AZURE_BLOB_REQUEST_TIMEOUT)`